### ISSUE TITLE
Resolve the column selection conversation

### ISSUE THREAD

**Issue by shreeve** [Microsoft/vscode#66444]
As a fairly new user of Visual Studio Code, and like apparently _many_ other users, I found column selection in `vscode` to be confusing and unnatural.

As I went to look for help in the issue tracker, I was blown away with how many problems and issues other users have had with this:

#1515 - Column selection
#2789 - Feedback: Missing Plugins
#3130 - Allow customization of mouse shortcuts
#5402 - Improved column selection (Alt+select)
#5851 - Give option to change the key used for multi-selection
#11830 - Column mode selection behavior is inconsistent with user expectations
#21785 - Not able to change keybinding for column select
#27193 - Allow for ctrl+click to create multiple cursors
#27480 - Test editor.multicursorModifier
#27503 - editor.multicursorModifier: cmd should be the default
#27680 - editor.multiCursorModifier ctrl/cmd settings are not portable between OS
#28186 - Column Selection by Middle Mouse Button (Sublime Style)
#36496 - Feature Request: Columns and rows selection like Visual Studio
#41728 - Right clicking column selection cancels selection
#46980 - Column selection in Ubuntu
#64661 - column select mode

I believe there are two small tweaks to resolve the problem in a natural and intuitive way:

* Set `editor.multiCursorModifier` to `ctrlCmd`
* Modify the code in `src/vs/editor/browser/view/viewController.ts` to:

``` patch
diff --git src/vs/editor/browser/view/viewController.ts src/vs/editor/browser/view/viewController.ts
index 0279da5670..e5f8873eb5 100644
--- src/vs/editor/browser/view/viewController.ts
+++ src/vs/editor/browser/view/viewController.ts
@@ -193,7 +193,11 @@ export class ViewController {
                                }
                        } else {
                                if (data.inSelectionMode) {
-                                       this.moveToSelect(data.position);
+                                       if (data.altKey) {
+                                               this.columnSelect(data.position, data.mouseColumn);
+                                       } else {
+                                               this.moveToSelect(data.position);
+                                       }
                                } else {
                                        this.moveTo(data.position);
                                }
```

With these changes:

* To use multiCursors, use `cmd+click`, which is more consistent with other editors anyway, instead of `alt+click`. This is possible due to `editor.multiCursorModifier` being set to `ctrlCmd`.
* To use columnSelect, use `alt+click` to *start* making a column selection at that position, which fixes the confusing `vscode` default column selection behavior of *extending* a column selection from the active cursor position prior to clicking with your mouse. This is possible due to the patch above.
* To use columnSelect, you can also use the default behavior of just making a column selection by using the middle mouse button when doing the selection. This is the default behavior.

The net effect is an intuitive key mapping that matches other common editors such as Sublime.

**Comment by shreeve** [comment]
See #66445

**Comment by shreeve** [comment]
@alexandrudima - Is this helpful?

**Comment by shreeve** [comment]
Closed via #66445
